.github/workflows/trigger.yml

```diff
@@ -14,29 +14,61 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  # Gate all builds through this check to prevent wasted runs.  Only run on
+  # 'labeled' events when the label is actually 'ci:main'.  Concurrency control
+  # above handles canceling the 'opened' event when 'labeled' arrives quickly
+  # after (e.g., when creating a PR with ci:main already attached). See #1154.
+  check-trigger:
+    if: |
+      startsWith(github.repository, 'kernelkit/') &&
+      (github.event_name != 'pull_request' ||
+       github.event.action != 'labeled' ||
+       github.event.label.name == 'ci:main')
```

Comment:
If the PR has multiple labels, will this really be correct? You removed this
` !contains(github.event.pull_request.labels.*.name, 'ci:main'`

Reply:
Good question! The github.event.label.name only contains the single label that triggered this specific 
labeled event, not an array of all labels.

When you add multiple labels to a PR (even before creating it), GitHub fires separate labeled events for
 each label. So if a PR has 5 labels, you get 5 separate labeled events, each with
github.event.label.name set to one label.

Our condition github.event.label.name == 'ci:main' correctly filters to only run when the ci:main label
event fires. The other label events are skipped.

The labels.*.name array (with the *) is only used in the bash script at line 42 for target selection,
where we need to check ALL labels on the PR. That works correctly with multiple labels.